feat(otel): Add docs for Ruby OpenTelemetry (#5856)
getsentry/sentry-docs: src/platform-includes/performance/opentelemetry-setup/ruby.mdx

> Initialize Sentry for tracing and set the `instrumenter` to `:otel`.

Reviewer: ```suggestion
Initialize Sentry for tracing and set the `instrumenter` to `:otel`:
```

> Initialize Sentry for tracing and set the `instrumenter` to `:otel`. Make sure that Sentry is initialized before the OpenTelemetry SDK is.

Reviewer 2: precedence order is no longer necessary, you can remove this line

```ruby {filename:config/initializers/sentry.rb}
Sentry.init do |config|
  config.dsn = "__DSN__"
  config.traces_sample_rate = 1.0
  # set the instrumenter to use OpenTelemetry instead of Sentry
  config.instrumenter = :otel
end
```

> This will disable all Sentry instrumentation and rely on the chosen OpenTelemetry tracers for creating spans.

Reviewer: Grammatically, this isn't quite correct. What will rely on OTEL? Your app? The SDK?
```suggestion
This disables all Sentry instrumentation and relies on the chosen OpenTelemetry tracers for creating spans.
```

Author: The OpenTelemetry SDK generates an spans that the Sentry SDK consumes. The Sentry SDK then takes those spans, transforms them into the Sentry format, and sends it to Sentry.

How does this sound?

```suggestion
This disables all Sentry performance instrumentation and uses the OpenTelemetry SDK to generate spans.
```

Reviewer: Without getting into the nitty gritty of the grammar, how about this:
```suggestion
This disables all Sentry performance instrumentation so that the Sentry SDK relies on the chosen OpenTelemetry tracers to generate spans.
```

> Next, configure OpenTelemetry as per your needs and hook in the Sentry span processor and propagator.

Reviewer: ```suggestion
Next, configure OpenTelemetry as you need and hook in the Sentry span processor and propagator:
```
